src/systems/a320_systems/src/hydraulic.rs

```diff
@@ -503,137 +556,146 @@ impl A320Hydraulic {
     // All the core hydraulics updates that needs to be done at the slowest fixed step rate
     fn update_fixed_step<T: Engine, U: EngineFirePushButtons>(
         &mut self,
-        context: &UpdateContext,
+        context_original_sim_delta: &UpdateContext,
         engine1: &T,
         engine2: &T,
         overhead_panel: &A320HydraulicOverheadPanel,
         engine_fire_push_buttons: &U,
         lgciu1: &impl LgciuInterface,
         lgciu2: &impl LgciuInterface,
     ) {
-        self.brake_computer.send_brake_demands(
-            &mut self.braking_circuit_norm,
-            &mut self.braking_circuit_altn,
-        );
+        // Then run fixed update loop for main hydraulics
+        while let Some(cur_time_step) = self.core_hydraulic_updater.next() {
```

Comment:
**praise**: I like using an `Iterator` for this. Good idea!
**suggestion**: The idea is great, but this doesn't yet really use the `Iterator` semantics in the sense that you could also add a `next` function to the `FixedStepLoop` without implementing `Iterator` and it would still work. I would suggest really implementing this throughout such that you end up with simply having a for loop over items coming from `self.core_hydraulic_updater`.

Reply:
Damn I thought that's what I did by implementing the iterator next() trait
"impl Iterator for FixedStepLoop "

You mean I should override more iterator default methods?

Reply:
So. I made it work using #[derive(Clone, Copy)] for iterators. 

It works but I'm not sure of what this implies under the hood there.

Reply:
This means it will copy the `FixedStepLoop` type every time you iterate over it, which isn't really what you want. I'll add a separate review with suggestions.